### DISCUSSION TITLE
Question about closing the lid

### DISCUSSION THREAD

**Discussion by acikul** [mhaeuser/Battery-Toolkit#1] [Q&A] [answered]
Hi, does the charge limit stay even if the Macbook is connected to the charger after closing the lid?
This is one of the drawbacks of using AlDente so I was wondering how it was implemented here.

**Comment by mhaeuser** [comment, marked as answer]
Hey, this usually works for me (in fact, never failed), but it cannot be guaranteed. Unlike AlDente, which seems to use timers for most things, this app uses XNU events for both AC connection status and charging progress. So, Battery Toolkit probably has a much bigger chance to get notified and disable sleep in time, but it remains a chance nevertheless.
